File: 217-contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        // int i;
        // Arrays.sort(nums);
        // for(i=0;i<nums.length-1;i++){
        //     if(nums[i]==nums[i+1])
        //         return true;
        // }
        // return false;

        // using hashset
        HashSet<Integer> h= new HashSet<Integer>();
        for(int i=0; i<nums.length; i++)
            if(h.contains(nums[i]))
                return true;
            else
                 h.add(nums[i]);
        return false;
           
        
    }
}

File: 217-contains-duplicate/contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        freq = {}
        for i in nums:
            if i not in freq:
                freq[i]=1
            else:
                freq[i]+=1
        for key, val in freq.items():
            if val > 1:
                return True
        return False